src/client/java/net/neoforged/neoforge/client/event/RegisterDebugEntriesEvent.java

```diff
@@ -0,0 +1,129 @@
+        this.entries = entries;
+        this.defaultProfile = defaultProfile;
+        this.performanceProfile = performanceProfile;
+    }
+
+    /**
+     * Registers a new {@link DebugScreenEntry} to allow it to be used for debug rendering and state toggling.
+     *
+     * @param id    Registration ID for this entry.
+     * @param entry Screen entry to be registered.
+     */
+    public void register(ResourceLocation id, DebugScreenEntry entry) {
+        if (entries.putIfAbsent(id, entry) != null)
+            throw new IllegalStateException("Duplicate DebugScreenEntry registration: " + id);
+    }
+
+    /**
+     * {@return true if the given entry id has been registered}
+     */
+    public boolean isRegistered(ResourceLocation id) {
+        return entries.containsKey(id);
+    }
+
+    /**
+     * Includes the entry into the given profile.
+     *
+     * @param id            Registration ID for this entry.
+     * @param profile       Debug profile this entry will be included with.
+     * @param profileStatus Status this entry will be set to when the profile is enabled.
+     */
+    public void includeInProfile(ResourceLocation id, DebugScreenProfile profile, DebugScreenEntryStatus profileStatus) {
+        if (getProfileMap(profile).putIfAbsent(id, profileStatus) != null)
+            throw new IllegalStateException("Duplicate DebugScreenEntry " + profile.getSerializedName() + "-profile inclusion: " + id);
+    }
+
+    /**
+     * {@return true if the entry is included into the given profile}
+     */
+    public boolean isIncludedInProfile(ResourceLocation id, DebugScreenProfile profile) {
+        return getProfileMap(profile).containsKey(id);
+    }
+
+    private Map<ResourceLocation, DebugScreenEntryStatus> getProfileMap(DebugScreenProfile profile) {
+        return switch (profile) {
+            case DEFAULT -> defaultProfile;
+            case PERFORMANCE -> performanceProfile;
+        };
+    }
+
+    @ApiStatus.Internal
+    public static void validateProfiles(RegisterDebugEntriesEvent event) {
+        // we delegate validation to its own method to allow people to call 'includeInProfile' before 'register'
+        var defaultError = validateProfile(event, DebugScreenProfile.DEFAULT);
+        var performanceError = validateProfile(event, DebugScreenProfile.PERFORMANCE);
+
+        // we delay throwing the error to allow both profiles to be validated in the same run
+        // we should still validate 'performance' if 'default' is invalid
+        if (defaultError != null && performanceError != null) {
+            var error = new IllegalStateException("Fatal errors occurred while validating DebugScreenProfiles");
+            error.addSuppressed(defaultError);
+            error.addSuppressed(performanceError);
+            throw error;
+        } else if (defaultError != null)
+            throw defaultError;
+        else if (performanceError != null)
+            throw performanceError;
```

Comment:
Please make the braces consistent, this mix of braces and no braces is awful to read